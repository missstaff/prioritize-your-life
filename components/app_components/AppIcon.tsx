import React from "react";
import { View } from "react-native";
import { ScaledSheet, s } from "react-native-size-matters";
import Ionicons from "@expo/vector-icons/Ionicons";
import { type IconProps } from "@expo/vector-icons/build/createIconSet";
import { type ComponentProps } from "react";

/**
 * Renders an app icon component.
 *
 * @param style - The style object for the component.
 * @param rest - Additional props for the component.
 * @returns The rendered app icon component.
 */
export function AppIcon({
  style,
  ...rest
}: IconProps<ComponentProps<typeof Ionicons>["name"] | any>): JSX.Element {
  return (
    <View style={[styles.container, style]}>
      <Ionicons size={s(28)} {...rest} />
    </View>
  );
}

const styles = ScaledSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: "transparent",
    borderRadius: 100,
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    padding: s(2.5),
  },
});
