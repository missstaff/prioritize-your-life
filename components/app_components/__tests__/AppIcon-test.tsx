import React from "react";
import renderer from "react-test-renderer";
import { AppIcon } from "../AppIcon";

describe("AppIcon Tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with default props", () => {
    const tree = renderer.create(<AppIcon name="home" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a custom size", () => {
    const tree = renderer.create(<AppIcon name="home" size={32} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a custom style", () => {
    const customStyle = { backgroundColor: "blue", padding: 10 };
    const tree = renderer
      .create(<AppIcon name="home" style={customStyle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with a different icon", () => {
    const tree = renderer.create(<AppIcon name="settings" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom color", () => {
    const tree = renderer.create(<AppIcon name="home" color="red" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
