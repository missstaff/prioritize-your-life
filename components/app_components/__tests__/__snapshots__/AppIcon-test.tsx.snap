// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon Tests renders correctly with a custom size 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 32,
        },
        undefined,
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`AppIcon Tests renders correctly with a custom style 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      {
        "backgroundColor": "blue",
        "padding": 10,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 60,
        },
        undefined,
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`AppIcon Tests renders correctly with custom color 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "red",
          "fontSize": 60,
        },
        undefined,
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`AppIcon Tests renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      undefined,
    ]
  }
>
  <Text />
</View>
`;

exports[`AppIcon Tests renders with a different icon 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 100,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 5.357142857142857,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 60,
        },
        undefined,
        {
          "fontFamily": "ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;
